File: mobile/src/main/java/com/ataulm/basic/MyAdapter.java
package com.ataulm.basic;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import java.util.List;

class MyAdapter extends RecyclerView.Adapter<MyViewHolder> {

    private final List<Item> items;
    private final MyViewHolder.Listener listener;

    public MyAdapter(List<Item> items, MyViewHolder.Listener listener) {
        this.items = items;
        this.listener = listener;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return MyViewHolder.createFrom(parent, listener);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

}

File: app/src/main/java/com/example/MyAdapter.kt
package com.example

import android.support.v7.widget.RecyclerView
import android.view.ViewGroup

class MyAdapter(private var list: List<Movie>) : RecyclerView.Adapter<MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder.inflate(parent)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.bind(list[position])
    }

    override fun getItemId(position: Int): Long {
        return list[position].id.toLong()
    }

    fun update(list: List<Movie>) {
        this.list = list
        // TODO: swap out for the new  ListAdapter
        notifyDataSetChanged()
    }
}
